### PR TITLE
Analyze nullability in implicit constructors

### LINKED ISSUE #26628: `null` initializer for non-nullable property (or field) should warn
Similar to https://github.com/dotnet/roslyn/issues/26626
```C#
public class C
{
    string x { get; set; } = null; // expect a warning
}
```

Found in nullable dogfood.

Note: there are some IDE tests (in `CSharpDeclareAsNullableCodeFixTests`) for this bug, which should be unskipped.

### LINKED ISSUE #31139: No warnings for field initializer with implicit constructor
A warning is reported for `A.x` but no warning is reported for `B.y`.
```c#
class A
{
    object x = F(null); // warning
    A() { }
    static object F(object o) => o;
}
class B
{
    object y = F(null); // no warning
    static object F(object o) => o;
}
```

### PR DESCRIPTION
**Pull request by cston**
Fixes #26628 
Fixes #31139

### CODE REVIEW

File: src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs [hunk 1]
```diff
@@ -193,7 +193,7 @@ protected override ImmutableArray<PendingBranch> Scan(ref bool badRegion)
             DiagnosticBag diagnostics,
             Action<BoundExpression, TypeSymbolWithAnnotations> callbackOpt = null)
         {
-            if (method.IsImplicitlyDeclared)
+            if (method.IsImplicitlyDeclared && (!method.IsImplicitConstructor || method.ContainingType.IsImplicitlyDeclared))
```
Comment (333fred) [inline src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs]: >method.ContainingType.IsImplicitlyDeclared [](start = 81, length = 42)

Which test is demonstrating this part of the check?

Comment (cston) [reply src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs]: This check prevents analyzing the implicit constructor of the (implicit) script class in particular. Without this check, several scripting tests fail within NullableWalker.

---
In reply to: [253556084](https://github.com/dotnet/roslyn/pull/33093#discussion_r253556084) [](ancestors = 253556084)

**Comment by cston** [comment]
@dotnet/roslyn-compiler for a second review, thanks.

File: src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs (hunk 1 above)
Comment (cston) [reply src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs]: In the script case, the constructor and containing type are both implicit and the `if` condition should be `true`.

---
In reply to: [253564880](https://github.com/dotnet/roslyn/pull/33093#discussion_r253564880) [](ancestors = 253564880,253562102,253556084)

**Pull request merged**